### PR TITLE
Add a new bot example.

### PR DESCRIPTION
**Pull request by Dositan**
## Summary
The kind of advanced example of inheriting `commands.Bot` with examples and comments provided.

This change does not fix an issue, but provides a specific example for beginners to get started quickly.

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: examples/custom_bot.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
+import asyncio
+import logging
+import os
+
+import aiohttp
+import discord
+from discord.ext import commands
+
+from .custom_context import MyContext
```
Comment (pikaninja) [inline examples/custom_bot.py:9]: this is never shown, commented or described

File: examples/custom_bot.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+# The preparing part.
+bot = MyBot(description='A testing bot.')
+
+
+@bot.command(name='modules')
+async def command_modules(ctx: MyContext) -> None:
+    """List all the available modules the bot has."""
+    # With a new `bot.list_modules` function, we can list
+    # all of the cogs located in the `./cogs` directory.
+    await ctx.send('The list of my modules: ' + ', '.join(ctx.bot.list_modules))
```
Comment (pikaninja) [inline examples/custom_bot.py:119]: list_modules is a static method, not a property, you still need to call it

File: examples/custom_bot.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        super().__init__(
+            command_prefix=_get_prefix,
+            activity=discord.Game(name='with commands.'),
+            allowed_mentions=discord.AllowedMentions(
+                everyone=False, roles=False, replied_user=False
+            ),
+            **kwargs
+        )
+        # LOOP-RELATED
+        self.loop = asyncio.get_event_loop()
```
Comment (pikaninja) [inline examples/custom_bot.py:64]: this should be set automatically in the `__init__`

File: examples/custom_bot.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        # single session which can be called from everywhere in the project.
+        # Learn more at: https://docs.aiohttp.org/en/stable/http_request_lifecycle.html#how-to-use-the-clientsession
+        self.session = aiohttp.ClientSession(loop=self.loop)
+
+    async def on_ready(self) -> None:
+        """Event that gets triggered everytime you successfully run the bot.
+
+        Called when the bot is done preparing the data received from Discord.
+        """
+        logger.info('Logged in as: {0} | ID: {0.id}'.format(self.user))
```
Comment (MrKomodoDragon) [inline examples/custom_bot.py:76]:
```suggestion
        logger.info(f’Logged in as: {self.user} | ID: {0.user’.)
```
Probably should use f-strings

**Review by pikaninja — COMMENTED** [review]
Couple nitpicks and issues

**Review by MrKomodoDragon — CHANGES_REQUESTED** [review]
A couple nits here and there

File: examples/custom_bot.py (hunk 2 above)
Comment (Dositan) [reply examples/custom_bot.py:119]: Right, thanks

File: examples/custom_bot.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+import discord
+from discord.ext import commands
+
+from .custom_context import MyContext
+
+__all__ = ('MyBot',)
+
+# Logging is the important part for all projects.
+# set the logger up for your own purposes.
+logger = logging.getLogger('bot')
```
Comment (Daggy1234) [inline examples/custom_bot.py:15]: You need to get the `discord` logger for accessing the library's logging?

https://discordpy.readthedocs.io/en/latest/logging.html#logging-setup

File: examples/custom_bot.py [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+
+def _get_prefix(bot: 'MyBot', message: discord.Message) -> str:
```
Comment (Daggy1234) [inline examples/custom_bot.py:18]: Why is the `MyBot`?

**Review by Daggy1234 — CHANGES_REQUESTED** [review]

**Comment by Rapptz** [comment]
Thanks for the PR. I think the current bot example is fine. Not sure whether this one's really adding anything that the others didn't.

File: examples/custom_bot.py [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+            ),
+            **kwargs
+        )
+        # LOOP-RELATED
+        self.loop = asyncio.get_event_loop()
+        # Creating a new session everytime you need to request
+        # is considered as a bad practice. Here, we are declaring a
+        # single session which can be called from everywhere in the project.
+        # Learn more at: https://docs.aiohttp.org/en/stable/http_request_lifecycle.html#how-to-use-the-clientsession
+        self.session = aiohttp.ClientSession(loop=self.loop)
```
Comment (Gobot1234) [inline examples/custom_bot.py:69]: The loop argument is deprecated, you shouldn't pass it

**Pull request closed without merging**